### ISSUE TITLE
Coroutines issue on runtime

### ISSUE THREAD

**Issue by ricoharisin** [uber/okbuck#839]
I'm trying to implement `coroutines` and okbuck can compile it with no problem.

but when i run the apps, it's crashed 

    java.lang.IllegalStateException: Module with the Main dispatcher is missing. Add dependency providing the Main dispatcher, e.g. 'kotlinx-coroutines-android'
                                                                      at kotlinx.coroutines.internal.MissingMainCoroutineDispatcher.missing(MainDispatchers.kt:73)
                                                                      at kotlinx.coroutines.internal.MissingMainCoroutineDispatcher.isDispatchNeeded(MainDispatchers.kt:54)
                                                                      at kotlinx.coroutines.DispatchedKt.resumeCancellable(Dispatched.kt:373)
                                                                      at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:25)
                                                                      at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:109)
                                                                      at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:160)
                                                                      at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:54)
                                                                      at kotlinx.coroutines.BuildersKt.launch(Unknown Source)
                                                                      at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch$default(Builders.common.kt:47)
                                                                      at kotlinx.coroutines.BuildersKt.launch$default(Unknown Source)

I already add `kotlinx-coroutines-android` dependency on module that use `coroutines` but the problem still exist 

It's working well if i compile it using gradle

okbuck v0.41
kotlin v1.3.11
coroutines v1.1.0
gradle v3.2.0

any ideas for this issue?

**Comment by kageiit** [comment]
Can you provide a small repro case/project?

**Comment by ricoharisin** [comment]
Thank you for your response, i have made sample project with coroutines 

https://github.com/ricoharisin/coroutines-sample

at first, i thought it was related to this issue https://github.com/facebook/buck/issues/1897 but turned out it's not the case because generated META-INF exist in  jar

**Comment by ricoharisin** [comment]
after inspecting the apk, i found at that generated apk not including META-INF/services file from coroutines android lib

**Comment by rifqimfahmi** [comment]
@kageiit I am also facing the same problem like @ricoharisin. when i inspect the `coroutinescope.launch` method on Builders.common.kt file, it somehow detect the CoroutineContext as instance of `MissingMainCoroutineDispatcher` class instead of `HandlerContext` when i install it with okbuck. On the other hand, installing it with gradle work as expected. I provide the screenshots of the debugging between using okbuck and gradle and detailed info to reproduce in the repo: https://github.com/rifqimfahmi/coroutines-sample

- Gradle Version: 5.1.1
- Kotlin: 1.3.20
- Coroutine: 1.1.0
- okbuck version: 0.48.4 (my internal version 0.48.5 to fix the AGP as stated in #841 ) you can see the repo at https://github.com/rifqimfahmi/okbuck
- Also I' m using buck binary [facebook/buck@91b30eb](https://github.com/facebook/buck/commit/91b30eb54128130f925c3d968619c5b59c888486) as stated by @raviagarwal7 in #798 

Thank you

**Comment by ricoharisin** [comment]
as @rifqimfahmi said, the main problem is generated service files from coroutines is not included in the final APK

after check the buck source code, it seems the `ApkBuilder` deliberately strip `META-INF` directory

for now i able to make it works by using dirty workaround using `apk_genrule` and add it in application `BUCK` file

    apk_genrule(
      name = 'bin_debug_mod',
      srcs = ['debug.keystore', 'META-INF'],
      apk = ':bin_debug',
      bash = 'unzip $APK -d bin_debug &&' \
            'cp -R META-INF/* bin_debug/META-INF &&' \
            '(cd bin_debug && zip -r ../mod_unaligned.apk .) &&' \
            '$ANDROID_SDK/build-tools/27.0.3/zipalign -v -p 4 mod_unaligned.apk $OUT &&' \
            '$ANDROID_SDK/build-tools/27.0.3/apksigner sign --ks debug.keystore --ks-pass pass:android $OUT',
      out = 'bin_debug_mod.apk'
    )

basically this script will unzip the apk, manually copy `META-INF` directory that comes from coruotines, repack and resigned the apk again

**Comment by rifqimfahmi** [comment]
This was fixed on [facebook/buck#2291](https://github.com/facebook/buck/pull/2291)
